### DISCUSSION TITLE
Cordova writing to persistent filesystem on browser platform

### DISCUSSION THREAD

**Discussion by rdindir** [apache/cordova#472] [General]
Hello

I am using the browser platform on cordova, using the File plugin. I am trying to write a file to the LocalFileSystem.PERSISTENT file store. Can I get to the actual file that is being written on the file system.

I am using linux.

Regards

**Comment by breautek** [comment]
> Can I get to the actual file that is being written on the file system.

Depends on what you mean by this. If you can read the actual file from outside the browser? Then no, the filesystem API on the browser is sandboxed. It uses the browser API if available (e.g. on Chrome) otherwise it fallsback to an IndexedDB implementation. Both implementations are sandboxed/virtual filesystems. No file is actually stored on the physical disk (in a traditional sense). Instead they are stored in some kind of local database as part of browser user data (similar to how cookies or local storage is stored).

You can use the filesystem APIs to read the file however from within your app.

Relevant Links:
- [More on Browser Quirks](https://github.com/apache/cordova-plugin-file?tab=readme-ov-file#browser-quirks)
- [Read File Example](https://github.com/apache/cordova-plugin-file?tab=readme-ov-file#read-a-file-)

**Reply by rdindir** [reply]
I see. I am using firefox, then it is an IndexedDB. Is it possible to remove the file from the sandboxed/virtual file system? Or would it be possible to get to the IndexedDB from within the browser (using web developer tools)? To just clear it.

**Reply by breautek** [reply]
Not really familiar with the firefox dev tools, but I would suspect they have something to explore that data.

Looks like they have a `Storage` tab:
![Screenshot from 2024-03-22 10-42-05](https://github.com/apache/cordova/assets/11200662/db1d159f-ba97-49e9-8c15-3c43965cdb7a)

**Reply by rdindir** [reply]
Thanks Norman. There really is an "Indexed DB" entry, and I did not see it.
